src/cpp/session/modules/SessionDataImportV2.R

```diff
@@ -13,21 +13,180 @@
 #
 #
 
-.rs.addJsonRpcHandler("preview_data_import", function(path, maxRows, maxCols = 100, maxFactors = 64)
+.rs.addFunction("assemble_data_import_name", function(dataImportOptions)
+{
+  dataName <- dataImportOptions$dataName
+
+  if (is.null(dataName) || identical(dataName, ""))
+    {
+      if (!is.null(dataImportOptions$importLocation))
+      {
+        locationName <- basename(dataImportOptions$importLocation)
+        if (length(locationName) > 0)
+        {
+          fileComponents <- unlist(strsplit(locationName, ".", fixed = TRUE))
+          components <- length(fileComponents)
+          if (components >= 1)
+          {
+            dataName <- paste(fileComponents[1:components-1], collapse = "_")
```

Comment:
So far so good, so my guess is the default precedence is `1:(component-1)`; otherwise I should be getting an error. Regardless, I'll make the change to be explicit here.